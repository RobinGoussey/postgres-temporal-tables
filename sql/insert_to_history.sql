CREATE EXTENSION IF NOT EXISTS psql_temporal;

CREATE TABLE insert_firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO insert_firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('insert_firemen_temporal');

-- test functionality of history
INSERT INTO insert_firemen_temporal (name,"rank") VALUES ('Bert','Cp.');


-- We can't use the timestamp columns, as they are constantly changing
select id , name , rank from temporal_insert_firemen_temporal_history where sysendtime is null and name = 'Bert';
