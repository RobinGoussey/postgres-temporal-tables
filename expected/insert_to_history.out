CREATE EXTENSION IF NOT EXISTS psql_temporal;
NOTICE:  extension "psql_temporal" already exists, skipping
CREATE TABLE insert_to_history_firemen_temporal (
	id serial PRIMARY KEY,
	name text,
	"rank" text
);
-- fill table
INSERT INTO insert_to_history_firemen_temporal (name,"rank") VALUES ('Bram','Sgt.');
-- create tables
select public.convert_to_temporal('insert_to_history_firemen_temporal');
NOTICE:  trigger "insert_to_history_firemen_temporal_temporal_trigger" for relation "public.insert_to_history_firemen_temporal" does not exist, skipping
 convert_to_temporal 
---------------------
 t
(1 row)

-- test functionality of history
INSERT INTO insert_to_history_firemen_temporal (name,"rank") VALUES ('Bert','Cp.');
-- We can't use the timestamp columns, as they are constantly changing
select id , name , rank from temporal_insert_to_history_firemen_temporal_history where sysendtime is null and name = 'Bert';
 id | name | rank 
----+------+------
  2 | Bert | Cp.
(1 row)

